protocol-units/settlement/mcr/contracts/script/StakingDeployer.s.sol

```diff
@@ -0,0 +1,75 @@
+pragma solidity ^0.8.13;
+
+import "forge-std/Script.sol";
+import {MovementStaking} from "../src/staking/MovementStaking.sol";
+import {TransparentUpgradeableProxy} from "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+import {TimelockController} from "@openzeppelin/contracts/governance/TimelockController.sol";
+import { Helper } from "./helpers/Helper.sol";
+
+contract StakingDeployer is Helper {
+
+    function run() external virtual {
+        
+         // load config and deployments data
+        _loadExternalData();
+
+        uint256 signer = vm.envUint("PRIVATE_KEY");
+        vm.startBroadcast(signer);
+
+        // Deploy CREATE3Factory, Safes and Timelock if not deployed
+        _deployDependencies();
+
+        deployment.stakingAdmin == ZERO && deployment.staking == ZERO && deployment.move != ZERO ?
+            _deployStaking() : deployment.stakingAdmin != ZERO && deployment.staking != ZERO ?
+                _upgradeStaking() : revert("STAKING: both admin and proxy should be registered");
+
+        vm.stopBroadcast();
+
+        // Only write to file if chainid is not running a foundry local chain
+        if (vm.isContext(VmSafe.ForgeContext.ScriptBroadcast)) {
+                _writeDeployments();
+            }
+    }
+
+    // •☽────✧˖°˖DANGER ZONE˖°˖✧────☾•
+
+    function _deployStaking() internal {
+        console.log("STAKING: deploying");
+        MovementStaking stakingImplementation = new MovementStaking();
+        vm.recordLogs();
+        stakingProxy = new TransparentUpgradeableProxy(
```

Comment:
Why not UUPS?

Reply:
UUPS is not as commonly implemented, especially for tokens. Since all of them should follow a similar workflow and there is one danger to UUPS:
![image](https://github.com/user-attachments/assets/ddc7bd1f-3309-4aed-9eeb-1fde793ecf08)